compiler/rustc_expand/src/proc_macro_server.rs

```diff
@@ -368,19 +370,17 @@ impl<'a, 'b> Rustc<'a, 'b> {
     pub fn new(ecx: &'a mut ExtCtxt<'b>) -> Self {
         let expn_data = ecx.current_expansion.id.expn_data();
         Rustc {
+            sess: ecx.parse_sess(),
             def_site: ecx.with_def_site_ctxt(expn_data.def_site),
             call_site: ecx.with_call_site_ctxt(expn_data.call_site),
             mixed_site: ecx.with_mixed_site_ctxt(expn_data.call_site),
+            span_debug: ecx.ecfg.span_debug,
             krate: expn_data.macro_def_id.unwrap().krate,
             rebased_spans: FxHashMap::default(),
             ecx,
         }
     }
 
-    fn sess(&self) -> &ParseSess {
-        self.ecx.parse_sess()
-    }
```

Comment:
maybe we'll get the same perf if `parse_sess` and `sess` are both `#[inline]`?

Reply:
Perhaps. My reasoning here was just that it would remove some pointer chasing, as what was previously 1 indirect read (`self.sess`) is now 2 (`self.ecx.sess`, `parse_sess` is an inline field of `Session`, so doesn't require an additional pointer chase). If the functions are not being inlined, though, that's more likely to be the cause of the perf loss.

I'll push another version where I just `#[inline]` the two functions.